pipenv/utils/funktools.py

```diff
@@ -62,18 +62,13 @@ def unnest(elem: Iterable) -> Any:
     """
 
     if isinstance(elem, Iterable) and not isinstance(elem, str):
-        elem, target = tee(elem, 2)
-    else:
-        target = elem
-    if not target or not _is_iterable(target):
-        yield target
-    else:
-        for el in target:
+        for el in elem:
             if isinstance(el, Iterable) and not isinstance(el, str):
-                el, el_copy = tee(el, 2)
-                yield from unnest(el_copy)
+                yield from unnest(el)
```

Comment:
This makes total sense to use recursion here. I wonder how it was not done before.